# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/DeleteNodeTransaction.cpp
# Commit: de32238c780f
# Full Hash: de32238c780f734d57628ec3957dd7065ea75db3
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2021-09-15 09:24:53
# Description:
#   Bug 1727840 - Make `DeleteNodeTransaction` allow to create its instance for non-editable text node which was created or modified by an editor r=m_kato
#   
#   Although we don't have a testcase, `HTMLEditor` may need to delete a text node
#   if it's created for composition.  If it's not allowed, users may see staying
#   composition string in disabled editor.
# ==============================================================================

diff -r 007ff7f371c9 -r de32238c780f editor/libeditor/DeleteNodeTransaction.cpp
--- a/editor/libeditor/DeleteNodeTransaction.cpp	Wed Sep 15 00:04:16 2021 +0000
+++ b/editor/libeditor/DeleteNodeTransaction.cpp	Wed Sep 15 00:20:19 2021 +0000
@@ -33,8 +33,17 @@
     : mEditorBase(&aEditorBase),
       mContentToDelete(&aContentToDelete),
       mParentNode(aContentToDelete.GetParentNode()) {
-  MOZ_DIAGNOSTIC_ASSERT_IF(aEditorBase.IsHTMLEditor(),
-                           HTMLEditUtils::IsRemovableNode(aContentToDelete));
+  MOZ_DIAGNOSTIC_ASSERT_IF(
+      aEditorBase.IsHTMLEditor(),
+      HTMLEditUtils::IsRemovableNode(aContentToDelete) ||
+          // It's okay to delete text node if it's added by `HTMLEditor` since
+          // remaining it may be noisy for the users.
+          (aContentToDelete.IsText() &&
+           aContentToDelete.HasFlag(NS_MAYBE_MODIFIED_FREQUENTLY)));
+  NS_ASSERTION(
+      !aEditorBase.IsHTMLEditor() ||
+          HTMLEditUtils::IsRemovableNode(aContentToDelete),
+      "Deleting non-editable text node, please write a test for this!!");
 }
 
 std::ostream& operator<<(std::ostream& aStream,